src/main/scala/net/ripe/rpki/publicationserver/store/fs/RrdpRepositoryWriter.scala

```diff
@@ -22,17 +22,28 @@ class RrdpRepositoryWriter(conf: AppConfig) extends Logging {
 
   private val fileAttributes = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--"))
 
-  def writeNewState(rootDir: String, serverState: ServerState, newNotification: Notification, snapshot: Snapshot): Try[Option[FileTime]] =    
+  def writeNewState(rootDir: String, serverState: ServerState, newNotification: Notification, snapshot: Snapshot): Try[Option[FileTime]] =
+    writeNextState(rootDir, serverState, newNotification, snapshot, () => ())
+
+  def writeNewStateWithNotificationDelay(rootDir: String, serverState: ServerState, newNotification: Notification, snapshot: Snapshot): Try[Option[FileTime]] =
+    writeNextState(rootDir, serverState, newNotification, snapshot, () => delayNotificationXml())
+
+  private def writeNextState(rootDir: String,
+                             serverState: ServerState,
+                             newNotification: Notification,
+                             snapshot: Snapshot,
+                             maybeDelay: () => Unit) = {
     Try {
       writeSnapshot(rootDir, serverState, snapshot)
-      delayNotificationXml()
+      maybeDelay()
```

Comment:
Could git github wonkiness on my phone, but I do not see this function in the diff. If it already exists, then all is fine.

Reply:
It's the parameter. It's either doing nothing or waiting, depending on where is called from. It's a little redundant, but just shorter than having a boolean.